include/core/rect.hpp

```
@@ -18,10 +22,51 @@ struct UIRectParams {
+
+  static UIConstraints horizontallyLose(float minHeight, float maxHeight) noexcept {
+    return {0.0f, INFINITY, minHeight, maxHeight};
+  }
+
+  static UIConstraints lose(float minWidth, float maxWidth, float minHeight, float maxHeight) noexcept {
+    return {minWidth, maxWidth, minHeight, maxHeight};
+  }
+
+  static UIConstraints fitted(float width, float height) noexcept { return {width, width, height, height}; }
 
   UIConstraints shrinkBy(float horizontal, float vertical) const noexcept {
```

Clang (14.0.6)

[warning] function 'shrinkBy' should be marked [[nodiscard]]

(modernize-use-nodiscard)